superset/reports/commands/create.py

```diff
@@ -0,0 +1,93 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import json
+import logging
+from typing import Any, Dict, List, Optional
+
+from flask_appbuilder.models.sqla import Model
+from flask_appbuilder.security.sqla.models import User
+from marshmallow import ValidationError
+
+from superset.commands.utils import populate_owners
+from superset.dao.exceptions import DAOCreateFailedError
+from superset.databases.dao import DatabaseDAO
+from superset.models.reports import ReportScheduleType
+from superset.reports.commands.base import BaseReportScheduleCommand
+from superset.reports.commands.exceptions import (
+    DatabaseNotFoundValidationError,
+    ReportScheduleAlertRequiredDatabaseValidationError,
+    ReportScheduleCreateFailedError,
+    ReportScheduleInvalidError,
+    ReportScheduleNameUniquenessValidationError,
+)
+from superset.reports.dao import ReportScheduleDAO
+
+logger = logging.getLogger(__name__)
+
+
+class CreateReportScheduleCommand(BaseReportScheduleCommand):
+    def __init__(self, user: User, data: Dict[str, Any]):
+        self._actor = user
+        self._properties = data.copy()
+
+    def run(self) -> Model:
+        self.validate()
+        try:
+            report_schedule = ReportScheduleDAO.create(self._properties)
+        except DAOCreateFailedError as ex:
+            logger.exception(ex.exception)
+            raise ReportScheduleCreateFailedError()
+        return report_schedule
+
+    def validate(self) -> None:
+        exceptions: List[ValidationError] = list()
+        owner_ids: Optional[List[int]] = self._properties.get("owners")
+        name = self._properties.get("name", "")
+        report_type = self._properties.get("type", ReportScheduleType.ALERT)
```

Comment:
Let's not assume that this is an alert unless otherwise specified - I think we should error if the front end fails to specify a type.

Reply:
done